packages/salesforcedx-apex-replay-debugger/src/adapter/apexReplayDebug.ts

```diff
@@ -514,6 +558,159 @@ export class ApexReplayDebug extends LoggingDebugSession {
     this.sendResponse(response);
   }
 
+  public replaceVariablesWithHeapDump(): void {
+    const topFrame = this.logContext.getTopFrame();
+    if (topFrame) {
+      const heapdump = this.logContext.getHeapDumpForThisLocation(
+        topFrame.name,
+        topFrame.line
+      );
+      if (heapdump && heapdump.getOverlaySuccessResult()) {
+        const frameInfo = this.logContext.getFrameHandler().get(topFrame.id);
+        const refVariableToRevisit = new Map<string, ApexVariableContainer>();
+        const valueVariableToRevisit = new Map<string, HeadpDumpExtentValue>();
+        for (const outerExtent of heapdump.getOverlaySuccessResult()!.HeapDump
+          .extents) {
+          if (topFrame.name.includes(outerExtent.typeName)) {
+            for (const innerExtent of outerExtent.extent) {
+              if (!innerExtent.address || !innerExtent.value.entry) {
+                continue;
+              }
+              const refContainer = this.logContext
+                .getRefsMap()
+                .get(innerExtent.address);
+              if (!refContainer) {
+                continue;
+              }
+              for (const variableEntry of innerExtent.value.entry) {
+                refContainer.variables.forEach((value, key) => {
+                  if (
+                    variableEntry.keyDisplayValue ===
+                    (value as ApexVariableContainer).name
+                  ) {
+                    if (
+                      typeof variableEntry.value.value === 'string' &&
+                      (variableEntry.value.value as string).startsWith('0x')
```

Comment:
Nitpick. I think instead of `.starsWith('0x')` it is better to refactor this into its own _intention revealing method_. See https://learn.co/lessons/intention-revealing-methods. 

In other words, we don't really care that it starts with 0x. What we care about is that it indicates that this could be a memory reference.